docs/architecture/cloud-native/grpc.md

```diff
@@ -0,0 +1,113 @@
+
+**Figure 4-20**. gRPC support in Visual Studio 2019
+  
+Figure 4-21 shows the skeleton gRPC service generated from the built-in scaffolding included in Visual Studio 2019.  
+
+![gRPC project in Visual Studio 2019](./media/grpc-project.png  )
+
+**Figure 4-21**. gRPC project in Visual Studio 2019
+
+In the previous figure, note the proto description file and service code. As you'll see shortly, Visual Studio generates additional configuration in both the Startup class and underlying project file.
+
+## gRPC usage
+
+Favor gRPC for the following scenarios:
+
+- Synchronous backend microservice-to-microservice communication where an immediate response is required to continue processing.
+- Polyglot environments that need to support mixed programming platforms.
+- Low latency and high throughput communication where performance is critical.
+- Point-to-point real-time communication - gRPC can push messages in real time without polling and has excellent support for bi-directional streaming.
+- Network constrained environments – binary gRPC messages are always smaller than an equivalent text-based JSON message.
+
+At the time, of this writing, gRPC is primarily used in backend services. Most modern browsers can't provide the level of HTTP/2 control required to support a front-end gRPC client. That said, there's an [early initiative](https://devblogs.microsoft.com/aspnet/grpc-web-experiment/) that enables gRPC communication from browser-based apps built with JavaScript or Blazor WebAssembly technologies. The  [gRPC-Web for .NET](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) enables an ASP.NET Core gRPC app to support gRPC features in browser apps:
+
+- Strongly-typed code-generated clients
+- Compact Protobuf messages
+- Server streaming
+
+## gRPC implementation
+
+The microservice reference architecture, [eShop on Containers](https://github.com/dotnet-architecture/eShopOnContainers), from Microsoft, shows how to implement gRPC services in .NET Core applications. Figure 4-22 presents the back-end architecture.
+
+![Backend architecture for eShop on Containers](./media/eshopWithAggregators.png)
+
+**Figure 4-22**. Backend architecture for eShop on Containers
+
+In the previous figure, note how eShop embraces the [Backend for Frontends pattern](https://docs.microsoft.com/azure/architecture/patterns/backends-for-frontends) (BFF) by exposing multiple API gateways. We discussed the BFF pattern in chapter 4. Pay close attention to the Aggregator microservice (in gray) that sits between the Web-Shopping API Gateway and backend Shopping microservices. The Aggregator orchestrates operations that require interaction with multiple backend services. These calls require synchronous communication and an immediate response. In eShop, backend calls from the Aggregator are performed using gRPC as shown in Figure 4-23.
+
+![gRPC in eShop on Containers](./media/grpc-implementation.png)
```

Comment:
The figure has a typo. Use `HTTP/2` instead of `Https/2`

Reply:
Great catch; fixed